spezi-module/contact/src/androidTest/java/edu/stanford/spezikt/spezi_module/contact/ContactScreenTest.kt

```diff
@@ -0,0 +1,109 @@
+package edu.stanford.spezikt.spezi_module.contact
+
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Call
+import androidx.compose.material.icons.filled.Email
+import androidx.compose.material.icons.filled.Info
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onNodeWithText
+import edu.stanford.spezikt.spezi_module.contact.model.Contact
+import edu.stanford.spezikt.spezi_module.contact.model.ContactOption
+import edu.stanford.spezikt.spezi_module.contact.model.ContactOptionType
+import edu.stanford.spezikt.spezi_module.contact.repository.ContactRepository
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.mockito.Mockito.mock
+import org.mockito.kotlin.whenever
+import java.util.UUID
+
+class ContactScreenTest {
+
+    private val mockContactRepository: ContactRepository = mock()
+
+    @get:Rule
+    val composeTestRule = createComposeRule()
+
+
+    @Before
+    fun setup() {
+        whenever(mockContactRepository.getContact()).thenReturn(createContact())
+    }
+
+    @Test
+    fun contactView_displaysContactName() {
+
+        composeTestRule.setContent {
+            ContactScreen(mockContactRepository.getContact())
+        }
+
+        composeTestRule.onNodeWithText(mockContactRepository.getContact().name).assertExists()
+    }
+
+
+    @Test
+    fun contactView_displaysContactOptions() {
+
+        composeTestRule.setContent {
+            ContactScreen(mockContactRepository.getContact())
+        }
+
+        mockContactRepository.getContact().options.forEach { option ->
+            composeTestRule.onNodeWithText(option.name).assertExists()
+        }
+    }
+
+    @Test
+    fun contactView_displaysContactTitle() {
+        composeTestRule.setContent {
+            ContactScreen(mockContactRepository.getContact())
+        }
+
+        composeTestRule.onNodeWithText(mockContactRepository.getContact().title).assertExists()
+    }
+
+    @Test
+    fun contactView_displaysContactDescription() {
+        composeTestRule.setContent {
+            ContactScreen(mockContactRepository.getContact())
+        }
+
+        composeTestRule.onNodeWithText(mockContactRepository.getContact().description)
+            .assertExists()
+    }
+
+    private fun createContact(): Contact {
+        return Contact(
```

Comment:
First, I would propose to always use parameter for every function and constructor - you can select the whole constructor here, right click, `Show context Actions` and `Add names to call arguments` - feel free to setup a shortcut on your device for this.

Next, since we might need objects in multiple tests, I usually go with the following patter. For every type, e.g. `Contact`, I would create a `ContactFactory` in the test module, which only exposes a single `create` method, with exactly the same parameter as the main constructor of the `Contact` object. All parameters have simple default values, e.g. empty string, zero, null and so on. And in every test you just create the object in an isolated manner and set the props relevant to the test only. E.g. `val contact = ContactFactory.create(name = "John Doe")`. And in the test you would just verify that the text node has indeed the expected contact name. Other props should be tested in separate tests.